delivery/services/delivery_service.py

```diff
@@ -82,7 +82,13 @@ def _create_links_area_for_project_runfolders(self, project_name, projects, batc
         for project in projects:
             try:
                 link_name = os.path.join(project_dir, project.runfolder_name)
-                self.file_system_service.symlink(project.path, link_name)
+                project_path = project.path
+                if (
+                    os.path.exists(runfolder_path := os.path.join(project.path, project.runfolder_name)) and 
+                    len(os.listdir(project_path)) == 1
+                ):
+                    project_path = runfolder_path
+                self.file_system_service.symlink(project_path, link_name)
```

Comment:
I see project.path does not seem to have the runfolder folder:

```
link_name
'/tmp/XYZ_123/132/160930_ST-E00216_0555_BH37CWALXX_utmbxd6p'

project_dir
'/tmp/XYZ_123/132'

project.path
'../arteria-delivery/tests/resources/runfolders/160930_ST-E00216_0555_BH37CWALXX_utmbxd6p/Projects/XYZ_123'
```

Reply:
@nkongenelly The project path does not point to a runfolder, but it does contain a folder named after the runfolder. Like we discussed, the organise service makes sure that the Projects folder looks like this: 
```
└── Projects
    └── ABC_123
         └── 180124_A00181_0019_BH72M5DMXX
                      └── test_file
```
In that example project path is `Projects/ABC_123` and it contains `180124_A00181_0019_BH72M5DMXX`. 

@Aratz I agree that we should be able to trust that the structure looks like that. This is how we reasoned when adding the conditional: The staging service's job is to stage everything in `ABC_123`  so that it will be delivered. If the organise requirements change or if we manually stage data in the projects folder, there is a risk that we unknowingly do not deliver all files in that folder, if we don't have the conditional. 

I do however see that we might be a bit over cautious here. If we would change anything in the organise service, we would do thorough testing. I guess the proper way to solve this would be to in a integration test between `organise` and `staging` would confirm that the staged data looks correct. 

Regarding the manual staging I mentioned, we actually do that quite often when staging Undetermined, but then we have clear instructions about how it should be done, and we do not add files directly in the projects folder (i.e. `Projects/AB_123`)  so that risk is minimal. 

TL;DR: I agree with @Aratz, we probably should skip the conditional here and just use `os.path.join(project.path, project.runfolder_name` as link target. But If we do we, should make sure that we have an integration test between `organise` and `staging` that confirms that the staged data looks correct.

Reply:
I agree, I think it's important to be able to trust how the data will look like, especially if it comes from a service we have control on. Otherwise  you'd never end adding if-statements that the data exists and that it is organized as you expect it to be. So yeah, the right way to prevent we don't break this by modifying organise or staging in the future is to add more integration tests :)

On a related note, when we harmonize this service, we should make sure to go through all the cases where you do things manually and see if it can be automated better, and possibly also see if there is a better way to organise runfolders.